src/screens/ForgotPasswordReset/index.tsx

```diff
@@ -0,0 +1,200 @@
+    const [successMessage, setSuccessMessage] = useState<string | undefined>('');
+
+    const navigation = useNavigation<any>();
+
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormProps>();
+  const [forgotPasswordResetMutation, { loading, error }] =
+    useForgotPasswordResetMutation({
+      onCompleted(res) {
+        if (res.confirmForgotPassword?.success === false) {
+          setErrorMessage('An error occurred during password reset.');
+        } else {
+          setSuccessMessage('Password reset successful. Rerouting to Login.');
+          setTimeout(() => {
+            navigation.navigate('LoginScreen');
+          }, 2500); 
+        }
+      }
+      });
+ 
+
+  const emailPattern = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
+
+  const onSubmit = async (data: ConfirmForgotPasswordInput) => {
+    const { email, password, confirmationCode } = data;
+
+    await forgotPasswordResetMutation({
+      variables: {
+        input: {
+          email,
+          password,
+          confirmationCode,
+        },
+      },
+    });
+  };
+  const getConfirmationCodeErrorMessage = (): string | undefined => {
+    if (errors?.confirmationCode) {
+      return 'Invalid confirmation code, please try again';
+    }
+    if (error instanceof ApolloError) {
+      return 'Invalid confirmation code, please try again';
+    }
+    return undefined;
+  };
+
+  return (
+    <Center flex={1} py="10">
+      <Box>
+        <Heading
+          size="lg"
+          fontWeight="600"
+          color="coolGray.800"
+          _dark={{
+            color: 'warmGray.50',
+          }}
+        >
+          Welcome to CraftPile!
+        </Heading>
+        <Heading
+          mt="1"
+          _dark={{
+            color: 'warmGray.200',
+          }}
+          color="coolGray.600"
+          fontWeight="medium"
+          size="xs"
+        >
+          Complete To Reset Password
+        </Heading>
+        <VStack space={3} mt="5">
+          <FormControl isInvalid={!!errors.email?.message} >
+            <FormControl.Label>Email ID</FormControl.Label>
+            <Controller
+              control={control}
+              name="email"
+              render={({ field: { onChange, onBlur, value } }) => (
+                <Input
+                  autoCapitalize="none"
+                  onChangeText={onChange}
+                  onBlur={onBlur}
+                  value={value}
+                />
+              )}
+              rules={{
+                required: { value: true, message: 'Email is required' },
+                pattern: {
+                  value: emailPattern,
+                  message: 'Invalid email address',
+                },
+              }}
+            />
+            {errors.email && (
+              <FormControl.ErrorMessage>
+                {errors.email.message}
+              </FormControl.ErrorMessage>
+            )}
```

Comment:
we shouldnt collect the email again, would be nice to just pluck this from context since they entered it on the previous screen. We can leave it here for now and create a follow up ticket to make those changes.

Reply:
Took out email input, implemented text so email carries over.